src/core/moves/SliceSamplingMove.cpp

```diff
@@ -14,8 +17,14 @@
 #include "RbOrderedSet.h"
 #include "StochasticNode.h"
 
+using boost::optional;
+
 using namespace RevBayesCore;
 
+const double log_0 = RbConstants::Double::min;
+
+// How can we make a run-time adjustable log level?
+int log_level = 0;
```

Comment:
I think these declarations would be better with the other members in the h file

Reply:
I think you need an `extern` declaration in the header file, and another declaration in the `.cpp` file.  Otherwise the symbol will be defined in multiple `*.o` files and the linker will give an error.

For the moment I wasn't planning to export the `log_0` symbol.  However, if other `.cpp` files wanted to use `log_0`, then should it defined somewhere more central than SliceSamplingMove.h?

Long-term, I was thinking of adding a type called log_double_t that stores the log of a number, but supports all the normal operators, like *, +, etc.  something like...

```
class log_double_t {
    /// Natural log of the number.
    double value;
public:

    /// Access the log of the number.
    constexpr double  log() const {return value;}
    /// Access the log of the number.
    constexpr double& log()       {return value;}

    log_double_t& operator +=(const log_double_t& y) {loginc(value,y.log()); return *this;}
    log_double_t& operator -=(const log_double_t& y) {value = logdiff(log(),y.log()); return *this;}
    constexpr log_double_t& operator *=(const log_double_t& y) {value += y.log(); return *this;}
    constexpr log_double_t& operator /=(const log_double_t& y) {value -= y.log(); return *this;}

    operator double() const {return exp(value);}

    constexpr log_double_t& operator=(double y) {
        assert(y >= 0);
        if (y == 0)
            value = log_0;
        else if (y == 1)
            value = 0;
        else
            value = ::log(y);
        return *this;
    }

    constexpr log_double_t():value(log_0) {}

    constexpr log_double_t(double x):value(0) {
        operator=(x);
    }
};

...
```

Reply:
I didn't get that these were intended to be general, sorry ! I thought they were members of the `SliceSamplingMove` class
since log_0 is directly referencing from RbConstants here I'm not sure if it's worth making a new symbol for it ? but if you think it is clearer that way I would think putting it in RbConstants as well is the way to go

I like the log_double idea but I think overloading the assignment operator would be confusing, I would use a different name to make it clear that there is a transformation happening (`setToLogOf(y)` ?)